pkg/catalog/catalog.go

```diff
@@ -104,49 +138,44 @@ func IsDevfileComponentSupported(devfile Devfile) bool {
 
 // ListDevfileComponents lists all the available devfile components
 func ListDevfileComponents() (DevfileComponentTypeList, error) {
-	var catalogDevfileList DevfileComponentTypeList
+	catalogDevfileList := &DevfileComponentTypeList{}
 	catalogDevfileList.DevfileRegistries = DevfileRegistries
 
+	devfileIndex := make([]DevfileIndexEntry, 0, 20)
```

Comment:
can we start with a much smaller capacity? do we even need to specify the capacity?

Reply:
The capacity is a guess at how many devfiles we'll need to process to avoid constantly resizing the slice when appending new elements. 20, while arbitrary, seems like a good value for the current registries. See https://golang.org/doc/effective_go.html#slices for more details.

Reply:
The problem with 20 is that if we have lesser number of index entries, the space will be wasted. But as you said, specifying the capacity will prevent constantly resizing the slice and 20 seems like a good value for now. Let's keep it then.